src/js/utils/string-utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Useful string handling utilities.
+ */
+define(['underscore'], function(_) {
+    'use strict';
+
+    var interpolateEscape, escapeInterpolateOptions, HTML;
+
+    /**
+     * Creates an HTML snippet.
+     *
+     * @param {string} text The HTML for the text.
+     * @constructor
+     */
+    function HTMLSnippet(text) {
+        this.text = text;
+    }
+    HTMLSnippet.prototype.toString = function() {
+        return this.text;
+    };
+
+    HTML = function(text) {
+        return new HTMLSnippet(text);
+    };
+
+    /**
+     * Returns an HTML-escaped text string by substituting values into
```

Comment:
I think HTMLSnippet needs to be external.  

We should agree on its name as well.  As you can see in my comment on the "[Implement in the platform ticket](https://openedx.atlassian.net/browse/UITK-75?focusedCommentId=176420&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-176420)", code outside of this method may want to check its type (String or HTMLSnippet), and determine whether or not it still needs to escape.